crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -107,7 +107,7 @@ enum TimelineFocusData<P: RoomDataProvider> {
 }
 
 #[derive(Clone, Debug)]
-pub(super) struct TimelineInner<P: RoomDataProvider = Room> {
+pub(super) struct TimelineController<P: RoomDataProvider = Room> {
```

Comment:
I don't like this naming, as it doesn't feel consistent with the rest of the SDK. `Controller` is really too generic and has no particular meaning. However, in this particular context, the `TimelineController` does a lot, it's really the `Timeline` implementation. I imagine the meaning for you is that it _controls_ the timeline. Well, all types control what it represents… I don't find a better name and I clearly understand why you don't like the `Inner` suffix.

I'm going to accept it because I don't have anything better to propose.

Reply:
Yeah, I think of it as the Controller in a MVC model — but even that is kind of wrong, as it should be more of a View Controller, really — and the model is scattered from multiple async data sources, the room provider, etc. so it's a bit unclear that there's a single model, and that MVC is a suitable metaphor for all of it.

Thanks, I'm happy to tweak the name later if we find a better one.